components/oauth/org.wso2.carbon.identity.oauth.ui/src/main/java/org/wso2/carbon/identity/oauth/ui/OAuthServlet.java

```diff
@@ -87,7 +87,7 @@ protected void service(HttpServletRequest req, HttpServletResponse resp)
                 reqToken = OAuthConstants.OAUTH_TOKEN + "=" + oauthToken + "&"
                         + OAuthConstants.OAUTH_TOKEN_SECRET + "=" + oauthTokenSecret + "&"
                         + OAuthConstants.OAUTH_CALLBACK_CONFIRMED + "=" + oauthCallbackConfirmed;
-                out.write(reqToken);
+                out.write(Encode.forUriComponent(reqToken));
```

Comment:
It's wrong to encode the full query string for percent encoding. That might break the uri. Instead encode only the untrusted query parameter values. That is oauthToken, oauthTokenSecret and oauthCallbackConfirmed